src/com/galactanet/gametable/IGametablePlugin.java

```diff
@@ -0,0 +1,10 @@
+package com.galactanet.gametable;
+
+/**
+ * All plugins should implement this interface.
+ * When the gametable is ready for the plugin to start, it will call the initialise method with the gametable instance.
+ */
+public interface IGametablePlugin {
+    void initialise(GametableFrame gametable);
```

Comment:
I think this is fine for now. But I am wondering if giving plugins free rein over the GametableFrame might make it quite easy for different plugins to conflict.

A rough idea I had was defining a set of "areas" where a plugin is allowed to interact. But the interaction is pulled from the plugin rather than the plugin injecting itself.
For example there is an area "LEFT PANEL", the plugin registers that it has a LEFT PANEL interaction. When the left panel is built all plugins which registered LEFT PANEL have their corresponding leftPanel method called. That method should return a swing component, all of the leftPanel plugin swing components are then added to the left panel. This also means if there is significant change to the left panel, it is unlikely to break existing plugins that use it because it is in charge of including them. Other plugins could also expose their own areas and then plugins could extend other plugins.

What do you think?